lib/spack/spack/build_environment.py

```diff
@@ -174,6 +174,14 @@ def clean_environment():
     for v in build_system_vars:
         env.unset(v)
 
+    # Unset mpi environment vars. These flags should only be set by
+    # mpi providers for packages with mpi dependencies
+    mpi_vars = [
+        'MPICC', 'MPICXX', 'MPIFC', 'MPIF70', 'MPIF90'
```

Comment:
```suggestion
        'MPICC', 'MPICXX', 'MPIFC', 'MPIF77', 'MPIF90'
```

Reply:
Apologies for resolving the prior thread while overlooking that suggestion.